File: frontend/types/task.ts
export type TaskPriority = "low" | "medium" | "high"

export interface Task {
  id: string
  title: string
  description?: string
  completed: boolean
  priority?: TaskPriority
  dueDate?: string // ISO date string
  createdAt: string
  updatedAt: string
  userId: string
}

export interface CreateTaskInput {
  title: string
  description?: string
  priority?: TaskPriority
  dueDate?: string
}

export interface UpdateTaskInput {
  title?: string
  description?: string
  priority?: TaskPriority
  dueDate?: string
  completed?: boolean
}

File: backend/models/task.go
package models

import "time"

type Task struct {
	ID          string     `json:"id"`
	Title       string     `json:"title"`
	Description string     `json:"description,omitempty"`
	Status      string     `json:"status"` // "todo", "in_progress", "done"
	Completed   bool       `json:"completed"`
	Priority    string     `json:"priority,omitempty"` // "low", "medium", "high"
	DueDate     *time.Time `json:"dueDate,omitempty"`
	CreatedAt   time.Time  `json:"createdAt"`
	UpdatedAt   time.Time  `json:"updatedAt"`
	UserID      string     `json:"userId,omitempty"`
}

type CreateTaskRequest struct {
	Title       string     `json:"title"`
	Description string     `json:"description,omitempty"`
	Status      string     `json:"status,omitempty"`
	Completed   *bool      `json:"completed,omitempty"`
	Priority    string     `json:"priority,omitempty"`
	DueDate     *time.Time `json:"dueDate,omitempty"`
}

type UpdateTaskRequest struct {
	Title       *string    `json:"title,omitempty"`
	Description *string    `json:"description,omitempty"`
	Status      *string    `json:"status,omitempty"`
	Completed   *bool      `json:"completed,omitempty"`
	Priority    *string    `json:"priority,omitempty"`
	DueDate     *time.Time `json:"dueDate,omitempty"`
}
